aten/src/ATen/native/Distributions.cpp

```diff
@@ -424,6 +424,10 @@ Tensor _dirichlet_grad_cpu(const Tensor& x, const Tensor& alpha, const Tensor& t
  */
 
 Tensor _s_binomial_cpu(const Tensor& count, const Tensor& prob, std::optional<Generator> gen) {
+  TORCH_CHECK(
+      count.scalar_type() == prob.scalar_type(),
+      "Binomial function arguments count and prob must have same datatype of type Float, got: count = ",
+      count.scalar_type(),", prob = ", prob.scalar_type());
```

Comment:
nit: how about something like to be consistent with multinomial's error checking
```
TORCH_CHECK(
      at::isFloatingType(count.scalar_type()),
      "binomial only supports floating-point dtypes for count, got: ",
      self.scalar_type());
TORCH_CHECK(
      at::isFloatingType(prob.scalar_type()),
      "binomial only supports floating-point dtypes for prob, got: ",
      self.scalar_type());
```